src/pages/settings/Profile/Contacts/ValidateContactActionPage.tsx

```diff
@@ -0,0 +1,68 @@
+import React, {useEffect, useRef} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import DotIndicatorMessage from '@components/DotIndicatorMessage';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as User from '@libs/actions/User';
+import Navigation from '@libs/Navigation/Navigation';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import ValidateCodeForm from './ValidateCodeForm';
+import type {ValidateCodeFormHandle} from './ValidateCodeForm/BaseValidateCodeForm';
+
+function ValidateContactActionPage() {
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const themeStyles = useThemeStyles();
+    const {translate} = useLocalize();
+    const validateCodeFormRef = useRef<ValidateCodeFormHandle>(null);
+    const [loginList] = useOnyx(ONYXKEYS.LOGIN_LIST);
+
+    const [pendingContactAction] = useOnyx(ONYXKEYS.PENDING_CONTACT_ACTION);
+    const loginData = loginList?.[pendingContactAction?.contactMethod ?? ''];
+
+    useEffect(() => {
+        if (!loginData || !!loginData.pendingFields?.addedLogin) {
+            return;
+        }
+
+        // Navigate to methods page on successful magic code verification
+        Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS.route);
+    }, [loginData, loginData?.pendingFields, loginList]);
+
+    useEffect(() => () => User.clearUnvalidatedNewContactMethodAction(), []);
```

Comment:
Once user existed the page, we should also clear the error (RBR) as well.
Let's also add `User.clearUnvalidatedNewContactMethodAction()` as the return function.

Reply:
Which function are you referring?

Reply:
Sorry. I mean the cleanup function of this useEffect.